src/rcc.rs

```diff
@@ -88,13 +90,22 @@ impl APB2 {
 const HSI: u32 = 8_000_000; // Hz
 
 pub struct CFGR {
+    hse: Option<u32>,
     hclk: Option<u32>,
     pclk1: Option<u32>,
     pclk2: Option<u32>,
     sysclk: Option<u32>,
 }
 
 impl CFGR {
+    pub fn hse<F>(mut self, freq: F) -> Self
```

Comment:
can you 
```rust
fn use_hse<H: Into<Hertz>>(self, hse_freq: H) -> Self
```
and add doc comments? I will then be OK to merge if @therealprof have no objection.

Reply:
OK, looks good (I presume "mut self"?), it'll have to wait for a while though as I only have my phone with me while traveling for a couple of weeks. Thanks for taking a look!

Reply:
Yes, mut self. No hurry, take your time :smile:

Reply:
I finally got around to this! Does it look good to you?